### PR TITLE
Update LineContainer.mdx

### PR DESCRIPTION
**Pull request by TheScarletPhoenix**
From USACO Guide Module Bounties

_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: content/6_Advanced/LineContainer.mdx
Comment (SansPapyrus683) [inline content/6_Advanced/LineContainer.mdx:158]: 1. doc comment explaining what this does
2. rename `H` to something more informative or at least `h` bc snake_case

File: content/6_Advanced/LineContainer.mdx
Comment (SansPapyrus683) [inline content/6_Advanced/LineContainer.mdx:106]: just use INT32_MAX for inf

File: content/6_Advanced/LineContainer.mdx
Comment (SansPapyrus683) [inline content/6_Advanced/LineContainer.mdx:109]:
```suggestion
```

File: content/6_Advanced/LineContainer.mdx
Comment (SansPapyrus683) [inline content/6_Advanced/LineContainer.mdx:131]:
```suggestion
```

File: content/6_Advanced/LineContainer.mdx
Comment (SansPapyrus683) [inline content/6_Advanced/LineContainer.mdx:177]: can you use postincrements instead of preincrements everywhere

File: content/6_Advanced/LineContainer.mdx
Comment (SansPapyrus683) [inline content/6_Advanced/LineContainer.mdx:174]:
```suggestion
```

File: content/6_Advanced/LineContainer.mdx
Comment (SansPapyrus683) [inline content/6_Advanced/LineContainer.mdx:103]: 5 hashtags seems a bit much...

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
finally, another one on this site that knows how to use classes

**Comment by SansPapyrus683** [comment]
image

also uh 💀 
you probably shoulda contacted us to claim the bounty first

File: content/6_Advanced/LineContainer.mdx
Comment (SansPapyrus683) [inline content/6_Advanced/LineContainer.mdx:116]: hey man you know the precommit bot is changing things to tabs for a reason right

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx]: so, what do u want me to change?

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx]: no like i just find it interesting how pre commit changed everything to tabs and yet you always change it back to spaces

**Comment by TheScarletPhoenix** [comment]
merge the pull request now

**Comment by SansPapyrus683** [comment]
calm down man i haven't reviewed the other parts
also you need two approvals

File: content/6_Advanced/LineContainer.mdx
Comment (SansPapyrus683) [inline content/6_Advanced/LineContainer.mdx:161]:
```suggestion
/** @return the intersection of a set of halfplanes */
```

**Comment by SansPapyrus683** [comment]
https://docs.google.com/document/d/1Dzj5uYNYNXe_E-_9-lSeX73654cGJk88PZ0kZCCmks0/edit

so we've just recently come up with this new module layout standard, if you could adjust your module to fit this that'd be great

File: content/6_Advanced/LineContainer.mdx
Comment (SansPapyrus683) [inline content/6_Advanced/LineContainer.mdx:36]: if there's only 1 example problem, no need for this header
also the rest of the module doesn't follow the format i gave

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx]: I removed the header since there's just one focus problem, and I also have the explanation and implementation sections, as detailed in the document. If you have any more suggestions, please specify them (bcz I think the module looks good now).

**Comment by SansPapyrus683** [comment]
btw reviewing these things do take time, so please don't get nervous if nothing happens in like a week or so

**Comment by SansPapyrus683** [comment]
image

this part should be one pound sign (and the rest one less as well)
please take a closer look at the module layout guide

**Comment by TheScarletPhoenix** [comment]
I know the document said one pound sign and one less for the ones following, but I was hesitant to change it because it was already there (not part of the module I created). But I'll change that too.

**Comment by SansPapyrus683** [comment]
i mean what's stopping you from changing the second half

**Comment by TheScarletPhoenix** [comment]
just made some final changes (to the second half as well).

so, review???

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: content/6_Advanced/LineContainer.mdx
Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx]: done

File: content/6_Advanced/LineContainer.mdx
Comment (SansPapyrus683) [inline content/6_Advanced/LineContainer.mdx:114]: wait i just realized
is there an oj you can submit this code to to test it or anything

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:114]: no, it's not from competitions/hackathons or anything. Just like an intro/explanation example problem.

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx:114]: so how can users know if their code is correct

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:114]: by checking the solution ???

**Comment by TheScarletPhoenix** [comment]
Hey
Just checking in
Approximately when will my PR be reviewed/merged?

**Comment by SansPapyrus683** [comment]
ok just requested some other ppl to review lol

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]
images are not loading - can you download and import them into the repository?

**Comment by TheScarletPhoenix** [comment]
i did. the images should load now.

File: content/6_Advanced/LineContainer.mdx
Comment (ryanchou-dev) [inline content/6_Advanced/LineContainer.mdx:69]: make sure that you import it under the images folder in 6_Advanced.
image

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]
if you need any help with importing images, lmk!

File: content/6_Advanced/LineContainer.mdx
Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx]: the images still aren't loading for some reason even though i added them into the 6_Advanced folder and here in the files in this pr.

File: content/6_Advanced/LineContainer.mdx
Comment (SansPapyrus683) [inline content/6_Advanced/LineContainer.mdx:132]: what this method do

File: content/6_Advanced/LineContainer.mdx
Comment (SansPapyrus683) [inline content/6_Advanced/LineContainer.mdx:143]: and this
(i mean add doc comments)
like
```cpp
/** @return something, idk lol */
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: content/6_Advanced/LineContainer.mdx
Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx]: nvm, i figured it out. the images are loading now.

File: content/6_Advanced/LineContainer.mdx
Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx]: i just added doc comments in the specified format.

File: content/6_Advanced/LineContainer.mdx
Comment (SansPapyrus683) [inline content/6_Advanced/LineContainer.mdx:140]: uuuhhh idt these are valid doc comment things lol
https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html
see here for valid tags

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:140]: oh ok lol just saw. i'll fix them.

**Comment by TheScarletPhoenix** [comment]
hey
just checking in again
is there anything else to change before merging? (everything is in format and the images are loading now)

File: content/6_Advanced/LineContainer.mdx
Comment (SansPapyrus683) [inline content/6_Advanced/LineContainer.mdx:40]: wait so is there a focus problem for this

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx:40]: wait i just took a look at your code
there's no main method?

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:40]: it's in c++?

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx:40]: no yeah but,, you know most of the code here is intended to be submitted to a problem for ac purposes
your code here has no main method, so

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:40]: this problem is not from any contests

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx:40]: then how can we make sure it's correct

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:40]: the provided solution is correct. it does work.

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx:40]: can you send a proof of ac on a problem that used it?

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:40]: sure, but this code does work for this problem, like official solution.
and acceptance for what test cases in what problem? there's none.
the example problem's in the module, for which this solution works.

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx:40]: > this problem

what problem?

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx:40]: which specific one?

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:40]: the one explained in this module with pictures

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx:40]: unless you have an actual problem with test cases i'm pretty leery of adding this

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:40]: here, how about this: https://www.codechef.com/problems/CHN02.
i can also add extra practice problems on this topic.

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx:40]: as long as the sol works on that

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:40]: proof that the solution works on this problem:
image

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:40]: also, my code does have a main method; i just looked.

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:40]: and once you merge this, i can add a bunch of practice problems i found (from actual contests with test cases) in the json file.

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx:40]: > also, my code does have a main method; i just looked.

what line in the file?

> and once you merge this, i can add a bunch of practice problems i found (from actual contests with test cases) in the json file.

why not add them now?

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:40]: bcz rn the current module is unstructured without proper sections for the problems for each topic. once this is merged, the problems section will be formatted correctly.

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx:40]: you can make the sections yourself my guy

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:40]: ik and i did in this pr. but idk exactly where the problems i add in the json file would show up in the current module bcz rn it has like 3 problem sections when it only needs 2 (one for half-plane, other for line-container). if this pr is merged, there will be 2 problem sections like there supposed to and then i can add the problems. look at the current module now and u'll see how disorganized it is.

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx:40]: why...would merging this turn it back into 2 problem sections?

also
> what line in the file?

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:40]: nvm, i guess i'll still add those problems. and nvm for that too; i thought i saw main, but nope didn't

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:40]: @SansPapyrus683 i added those problems for half-plane intersection in the json file in this pr #4577

**Comment by TheScarletPhoenix** [comment]
now that i've added everything, merge???

**Comment by SansPapyrus683** [comment]
my guy you still have not added the main method

**Comment by TheScarletPhoenix** [comment]
i just added the main method

**Comment by SansPapyrus683** [comment]
yeah....you can change multiple files in a pr my guy

File: content/6_Advanced/LineContainer.mdx
Comment (SansPapyrus683) [inline content/6_Advanced/LineContainer.mdx:200]: bruh
what problem does this ac on

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:200]: the example one

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx:200]: no most modules give an algorithm _with_ a cp-style problem

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:200]: so, now should i add that problem and its solution in the module as well (the one i submitted proof of ac for)?

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:200]: but the solution still outputs what the algorithm is supposed to do; it gives the intersection points. so a cp problem doesn't matter??? the problem explanation in the module is still a great problem.

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx:200]: the whole point of usaco guide is to give an actual problem to apply these algorithms to in both examples and exercises

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:200]: so should i add practice problems

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx:200]: yes
i thought this was well established 💀

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:200]: and should i replace that ac problem with its new solution in place of the existing one? or should i add below the existing explanation and all?

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx:200]: replace

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx:200]: oh btw can i see this new problem? it has to be an extremely straightforward application of the algorithm you describe- just like the other cses problems the guide showcases

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:200]: yeah, the same one i got ac for:
https://www.codechef.com/problems/CHN02

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:200]: also idk how to add another file to this pr, so can i add that focus problem and other practice problems in the json file in a separate pr? as in reopen that closed pr...

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx:200]: i
how do you not know
how to commit multiple files

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx:200]: but yeah the problem seems nice

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:200]: ok i'll figure it out.

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:200]: it automatically makes a different pr, i tried. so can you reopen that closed pr? then, i can also add the new ac cp-style focus problem, in addition to the practice problems i already committed there.

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx:200]: oh.
are you working on the usaco guide editor?
please just...git clone it locally

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:200]: no, i'm working on the web editor on github.
idk does the same thing?
so reopen that pr, things will be much easier and faster?

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx:200]: no it's very limiting
just git clone

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:200]: ok
edit: i tried but nope it doesn't work, still requires some merging that creates a diff pr.
so rather i'll make another pr for the json file myself, like before. i don't see how it's limiting; moreover, it's more flexible.

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx:200]: i have never heard of anyone having to make two separate prs for two separate files
are you sure you're on the right branch?

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:200]: yeah

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:200]: yeah
so, i'm merging that closed pr's branch of added problems into this pr's branch in my forked repository, but it still makes a diff pr for merging that into this. so, rather i'll make a diff pr myself to separately update it.
bcz this is taking much longer than i expected, like so close to a 100 conversations, i've never seen that number in any pr.
reopening that is much easier and definitely faster.

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx:200]: > can you tell me what exactly you're doing, in terms of commits, pushes, whatever?
maybe a transcript of what shows on your terminal would help as well

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:200]: uh, i just did?

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:200]: i'll add the ac cp-style problem now

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:202]: update: i added the focus problem

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx:202]: where is it

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:202]: right after the resources and before the explanation

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:202]: i added it as a focus problem like in other modules, followed by explanation and solution.

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:202]: i also added the practice problems

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx:202]: the main method doesn't seem to read any input from stdin

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx]: i modified it. now it should work.

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx]: @SansPapyrus683 is there anything else to be reviewed and modified?

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx]: uh overall it looks fine, but the explanation needs to be looked at and i'm a bit busy lol
sorry for the inconvenience

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx]: oh it's ok

File: content/6_Advanced/LineContainer.mdx
Comment (SansPapyrus683) [inline content/6_Advanced/LineContainer.mdx:37]: > We can also state that a resulting intersection is always either bounded or empty

uuuhh what

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:37]: maybe in complex theory, but definitely not practically.

Comment (SansPapyrus683) [reply content/6_Advanced/LineContainer.mdx:37]: it takes like 0 effort to come up with an infinite intersection

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:37]: ik but...

Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx:37]: doesn't show up in cp?

**Comment by TheScarletPhoenix** [comment]
does this count as my second pr?

**Comment by SansPapyrus683** [comment]
no yeah this counts for your second pr

**Comment by TheScarletPhoenix** [comment]
no yeah i got that.
so merge this???

**Comment by SansPapyrus683** [comment]
i'm the only one who's really touched ur pr
we need another person to come around
who knows how long that'll take lol

**Comment by TheScarletPhoenix** [comment]
oh, yeah

**Comment by TheScarletPhoenix** [comment]
so far, didn't even get one approval.

File: content/6_Advanced/LineContainer.mdx
Comment (TheGamingMousse) [inline content/6_Advanced/LineContainer.mdx:187]: braces

**Review by TheGamingMousse — APPROVED** [review]
code looks good

File: content/6_Advanced/LineContainer.mdx
Comment (TheScarletPhoenix) [reply content/6_Advanced/LineContainer.mdx]: added

**Comment by TheScarletPhoenix** [comment]
@SansPapyrus683 @ryanchou-dev review???

File: content/6_Advanced/LineContainer.mdx
Comment (ryanchou-dev) [inline content/6_Advanced/LineContainer.mdx:84]:
```suggestion
* Sort the set of half-planes by angle; it takes $\mathcal{O}(N\logN)$ time.
```
use latex log

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]
Why did you save your diagrams in the root folder? Move your images to the section folder and import via relative paths.

**Comment by TheScarletPhoenix** [comment]
i moved all the images to the correct place and added them via relative paths

**Pull request closed without merging**
